react/new/src/types/theme.ts

```diff
@@ -0,0 +1,45 @@
+export type ThemeMode = 'light' | 'dark'
+
+export interface TeamColors {
+  primary: string
+  secondary: string
+}
+
+export type NHLTeam =
+  | 'ANA'
```

Comment:
Is this just a Union type like std::variant in cpp?

Reply:
Essentially yes, it's a union type. My C++ is pretty bad but I'm somewhat sure the concept is very similar in so much as they both represent a type that can hold several possible values.

Reply:
Yeah -- it's a new standard library thing in c++17, so relatively recent, but I've seen custom implementations before.